src/components/LayoutDialog.vue

```diff
@@ -83,19 +87,116 @@ function handleImageUpload(event: Event) {
-          image: e.target.result as string,
-          x: layout.backgroundImage?.x || 0,
-          y: layout.backgroundImage?.y || 0,
-          width: layout.backgroundImage?.width || 10,
-          height: layout.backgroundImage?.height || 10,
+        const map_image = new Image();
+        map_image.onload = async () => {
+          // get real width and height
+          const imageWidth = map_image.naturalWidth;
+          const imageHeight = map_image.naturalHeight;
+          // populate layout background image
+          layout.backgroundImage = {
+            image: map_image.src,
+            x: layout.backgroundImage?.x || 0,
+            y: layout.backgroundImage?.y || 0,
+            width: layout.backgroundImage?.width || imageWidth,
+            height: layout.backgroundImage?.height || imageHeight,
+            natural_width: imageWidth,
+            natural_height: imageHeight,
+          };
+          // update metadata from input if one was loaded before the map
+          const metadata_input = document.querySelector('#mapMetadata');
+          if (metadata_input) {
+            metadata_input.dispatchEvent(new Event('change'));
+          }
         };
+        map_image.src = e.target.result as string;
+
+        // TODO: if there is a metadata file in the same folder, it should be loaded it here
+        // loadMapMetadata(metadata_file);
       }
     };
-
     reader.readAsDataURL(file);
   }
 }
+
+function handleMetadataUpload(event: Event) {
+  const input = event.target as HTMLInputElement;
+  if (input.files && input.files[0]) {
+    const metadata_file = input.files[0];
+    loadMapMetadata(metadata_file);
+  }
+}
+
+function loadMapMetadata(metadata_file: File) {
+  if (!layout.backgroundImage?.image) {
+    return;
+  }
+  if (metadata_file) {
+    const reader = new FileReader();
+    reader.onload = () => {
+      try {
+        const yamlText = reader.result as string;
+        const parsedData = load(yamlText);
+        const mapMetadata: MapMetadata = parsedData as MapMetadata;
+
+        // Extract resolution and origin from the parsed data
+        if (
+          layout?.backgroundImage &&
+          mapMetadata?.resolution &&
+          mapMetadata?.origin
+        ) {
+          layout.backgroundImage.x = mapMetadata.origin[0];
+          layout.backgroundImage.y = mapMetadata.origin[1];
+          layout.backgroundImage.width =
+            mapMetadata.resolution * layout.backgroundImage.natural_width;
+          layout.backgroundImage.height =
+            mapMetadata.resolution * layout.backgroundImage.natural_height;
+        } else {
+          console.error(
+            'Failed to extract resolution and origin from map metadata.',
+          );
+        }
+      } catch (error) {
+        console.error('Error parsing YAML map metadata file:', error);
+      }
+      updateInputsFromLayout();
+    };
+    reader.readAsText(metadata_file);
+  }
+}
+
+function updateInputsFromLayout() {
```

Comment:
There is something that still bothers me: the `Input` don't update on their own when the v-model values are set...
I added the function `updateInputsFromLayout()` to force the update, but that seems redundant with the whole "reactive" variable concept.
@bekirbostanci Is there a way to auto-update without doing this kind of hack?

Reply:
Did you try to get layout backgroudImage text values ? This should work for you :) 

`props.layout.backgroundImage.value.x `

Reply:
I tried the following:
```js
function updateInputsFromLayout() {
  if (!layout?.backgroundImage) {
    console.log(
      'updateInputsFromLayout(): Layout has no backgroundImage field',
    );
    return;
  }
  // update inputs
  props.layout.backgroundImage.value.x = layout.backgroundImage.x;
  props.layout.backgroundImage.value.y = layout.backgroundImage.y;
  props.layout.backgroundImage.value.width = layout.backgroundImage.width;
  props.layout.backgroundImage.value.height = layout.backgroundImage.height;
}
```

However, it doesn't update the actual UX component.

Reply:
So I tried both 
```js
x_input.value = props.layout.backgroundImage.value.x;
```

and, because you never know,
```js
props.layout.backgroundImage.value.x = layout.backgroundImage.x;
```

However, it doesn't actually update the UX component.

Reply:
Yes, it does the trick, thanks